Relax ram requirements on kernel tests.

nashif opened the pull request:
Many of the tests had some old ram requirements, resetting all tests to not have any requirements.

nashif committed:
tests: kernel: remove ram restrictions

Remove all ram restrictions in kernel tests and revisit all tests and
try to make them pass on all platforms.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: kernel: common: rename test boot_delay

rename boot_delay function name for clarity and change doxygen group to
be more generic and part of the init group.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

andyross reviewed (approved):
IIRC I've added stuff like that in the past because of oddball platforms that fail to link.   But no point in keeping if we've retired those boards or whatever.

ioannisg reviewed (commented):
I am OK with this as long as it has been tested that the samples and tests build for all intended platforms.

ioannisg commented on tests/kernel/fifo/fifo_timeout/prj.conf:
```diff
@@ -1,3 +1,5 @@
 CONFIG_ZTEST=y
 CONFIG_IRQ_OFFLOAD=y
 CONFIG_MP_NUM_CPUS=1
+CONFIG_TIMESLICING=n
```
@nashif what was the reason for this?

nashif replied:
trying to save some ram :)

ioannisg replied:
sure, i wonder though if this affects the behavior of the test, or limits its coverage. Otherwise, no complaints.

nashif replied:
no it does not, this is a fifo object test, not scheduler test. Slicing is tested somewhere else and everywhere, as it is enabled by default.

nashif commented:
cortex_m0 test failing are real issues, no need to retrigger this in CI.

ioannisg commented:
@nashif going to check the qemu m0 failures

andrewboie commented on tests/kernel/lifo/lifo_usage/testcase.yaml:
```diff
@@ -1,5 +1,4 @@
-    min_ram: 32
     tags: kernel
-    platform_exclude: m2gl025_miv
+    #platform_exclude: m2gl025_miv
```
why comment out instead of remove?

andrewboie commented:
I think this needs to be validated with `sanitycheck --all`

nashif commented:
> I think this needs to be validated with `sanitycheck --all`

this already happens in CI, all changed tests/samples will run with all boards.

nashif commented:
> @nashif going to check the qemu m0 failures

@ioannisg  any progress on this?

nashif commented:
@ioannisg  any update on cortex_m0?

ioannisg commented:
> @ioannisg any update on cortex_m0?

Ups, sorry @nashif  didn't pick this task yet, but will do for v2.3 - is this needed before freeze date?

nashif commented:
> Ups, sorry @nashif didn't pick this task yet, but will do for v2.3 - is this needed before freeze date?

no

nashif committed:
tests: common: skip bootdelay test on qemu_cortex_m0

Test fails on this one platform, to unblock other changes, skip the test
while the issues is being looked at. This test never ran on this
platform before due to ram restrictions.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
